dp_wizard/app/__init__.py

```diff
@@ -31,13 +34,76 @@ def ctrl_c_reminder():  # pragma: no cover
     print("Session ended (Press CTRL+C to quit)")
 
 
+def _make_demo_csv(path: Path, contributions):
+    """
+    >>> import tempfile
+    >>> from pathlib import Path
+    >>> import csv
+    >>> with tempfile.NamedTemporaryFile() as temp:
+    ...     _make_demo_csv(Path(temp.name), 10)
+    ...     with open(temp.name, newline="") as csv_handle:
+    ...         reader = csv.DictReader(csv_handle)
+    ...         reader.fieldnames
+    ...         rows = list(reader)
+    ...         rows[0].values()
+    ...         rows[-1].values()
+    ['student_id', 'class_year', 'hw_number', 'grade', 'self_assessment']
+    dict_values(['1', '2', '1', '82', '0'])
+    dict_values(['100', '2', '10', '78', '0'])
+    """
+    random.seed(0)  # So the mock data will be stable across runs.
+    with path.open("w", newline="") as demo_handle:
+        fields = ["student_id", "class_year", "hw_number", "grade", "self_assessment"]
+        writer = csv.DictWriter(demo_handle, fieldnames=fields)
+        writer.writeheader()
+        for student_id in range(1, 101):
+            class_year = int(_clip(random.gauss(2, 1), 1, 4))
+            for hw_number in range(1, contributions + 1):
+                # Older students do slightly better in the class,
+                # but each assignment gets harder.
+                mean_grade = random.gauss(90, 5) + class_year * 2 - hw_number
+                grade = int(_clip(random.gauss(mean_grade, 5), 0, 100))
+                self_assessment = 1 if grade > 90 and random.random() > 0.1 else 0
+                writer.writerow(
+                    {
+                        "student_id": student_id,
+                        "class_year": class_year,
+                        "hw_number": hw_number,
+                        "grade": grade,
+                        "self_assessment": self_assessment,
+                    }
+                )
+
+
+def _clip(n: float, lower_bound: float, upper_bound: float) -> float:
+    """
+    >>> _clip(-5, 0, 10)
+    0
+    >>> _clip(5, 0, 10)
+    5
+    >>> _clip(15, 0, 10)
+    10
+    """
+    return max(min(n, upper_bound), lower_bound)
+
+
 def make_server_from_cli_info(cli_info: CLIInfo):
     def server(input: Inputs, output: Outputs, session: Session):  # pragma: no cover
-        public_csv_path = reactive.value(cli_info.public_csv_path or "")
-        private_csv_path = reactive.value(cli_info.private_csv_path or "")
+        if cli_info.is_demo:
```

Comment:
This information used to be provided by the `CLIInfo`, but I think it works better to have these details specified here.